typing/typemod.ml

```diff
@@ -1499,7 +1500,10 @@ and transl_signature ?(toplevel = false) env sg =
             in
             let pres =
               match tmty.mty_type with
-              | Mty_alias _ -> Mp_absent
+              | Mty_alias p ->
+                  if Env.is_functor_arg p env then
+                    raise (Error (pmd.pmd_loc, env, Cannot_alias p));
+                  Mp_absent
```

Comment:
I have been wondering if we want to the check this property in `transl_modtype` when creating the alias, but I think that raising the error here gives a better location : the whole `module M = Aliased` binding rather than just `Aliased`.